### PR TITLE
Cloudevents support for Segmentio/kafka-go library 

### PR DESCRIPTION
**Pull request by sravya-a**
Added support for segmentio similar to sarama library support for this issue: https://github.com/cloudevents/sdk-go/issues/683

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,4 +34,5 @@ _site/
 test/benchmark/e2e/http/results/
 tmp/
 test/coverage.tmp
-
+protocol/kafka_sarama/v2/vendor
+protocol/kafka_segmentio/v2/vendor
```
Comment (wuhuizuo) [inline .gitignore:38]: why?

File: protocol/kafka_segmentio/v2/message_benchmark_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package kafka_segmentio_test
```
Comment (wuhuizuo) [inline protocol/kafka_segmentio/v2/message_benchmark_test.go:6]: should it keep same with source files?

File: .gitignore (hunk 1 above)
Comment (n3wscott) [reply .gitignore:38]: perhaps the author ran `go mod vendor`? I agree, we don't need to commit the vendor dir nor ignore it.

File: protocol/kafka_segmentio/v2/go.mod [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module github.com/cloudevents/sdk-go/protocol/kafka_segmentio/v2
+
+go 1.15
+
+replace github.com/cloudevents/sdk-go/v2 => ../../../v2
+
+require (
+	github.com/cloudevents/sdk-go/v2 v2.0.0
```
Comment (n3wscott) [inline protocol/kafka_segmentio/v2/go.mod:8]: might want this to start as 2.5.0

File: protocol/kafka_segmentio/v2/message.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+	return &Message{
+		Value:       value,
+		ContentType: contentType,
+		Headers:     headers,
+	}
+}
+
+func (m *Message) ReadEncoding() binding.Encoding {
+	if m.version != nil {
```
Comment (n3wscott) [inline protocol/kafka_segmentio/v2/message.go:84]: version is nil for structured encoding in kafka?

File: protocol/kafka_segmentio/v2/message_benchmark_test.go (hunk 2 above)
Comment (n3wscott) [reply protocol/kafka_segmentio/v2/message_benchmark_test.go:6]: this is magic test package stuff for go, it means this test is not allowed to access private member functions or internal struct fields but it is allowed to live in the same folder as the source.

File: protocol/kafka_segmentio/v2/sender_test.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+package kafka_segmentio
+
+import (
+	"testing"
+)
+
+func TestSenderWithKey(t *testing.T) {
+	// topic := "aaa"
+	// writer := kafka.NewWriter(kafka.WriterConfig{
```
Comment (n3wscott) [inline protocol/kafka_segmentio/v2/sender_test.go:14]: any way to make this work in memory? or perhaps it should be an e2e test and we can launch the broker like we do in the other tests?

**Review by n3wscott — COMMENTED** [review]
Looking real good!! I am excited to have a second impl for a kafka client in the sdk. 

Looks like you need to sign your commits to pass DCO and some minor issues with formatting. 

Thanks for the hard work!!

**Comment by embano1** [comment]
@sravya-a is this PR still active?

**Comment by duglin** [comment]
@sravya-a did you still want this one merged? If so, can you please sign you commits

**Comment by duglin** [comment]
ping @sravya-a

**Comment by duglin** [comment]
@sravya-a I'm going to close this one, unless you'd like to wrap it. Please let me know

**Comment by duglin** [comment]
No response, closing. Ping if you'd like to reopen it

**Pull request closed without merging**